### ISSUE TITLE
Problem with Docs

### ISSUE THREAD

**Issue by rpspringuel** [gregorio-project/gregorio#1188]
In working on the installation over TeXLive 2016 issues, I've noticed a problem with the docs which I don't understand.

```
GregorioRef Warning: missing GABC for notdeferror: ...xmf-dist/tex/luatex/luaotfload/fontloader-2016-06-16.lua:15598: attempt to index field 'spacing' (a nil value)
.
<template> ...nd \hskip \z@ plus.5fill\relax \egroup \egroup 
                                                  \begingroup \CT@setup \CT@...

l.54 ...tTable{'greciliae','gregorio','granapadano'}
```

If I switch back to TL2015, then the error goes away, so the error is almost certainly related to TL2016.  Can anyone here further narrow this down so that I can report it in the right place?

**Comment by eroux** [comment]
this kind of bugs is certainly in luaotfload, maybe a good start would be to install the latest version... In case it doesn't work, I think you can report the bug with:

- the way to reproduce it (from scratch)
- your version of luatex
- your version of luaotfload

on [luaotfload's tracker](https://github.com/lualatex/luaotfload/issues)... This would help quite a lot if you manage to make the example as minimal as you can though...

**Comment by rpspringuel** [comment]
According to `tlmgr` I'm using the most recent version (at least it finds no available updates on CTAN).

Also looking closer at my log, it appears that the problem may be in the font table Appendix.  I'll try to narrow it down as best I can, but I've not delved into that code before, so if I could get some help there I'd appreciate it.

**Comment by henryso** [comment]
I'll fix it, but I have to install TL 2016 first.

**Comment by henryso** [comment]
There are two problems here.  The first is `missing GABC for notdeferror`.  This can be fixed by adding
```
  ['notdef'] = true,
  ['nonmarkingreturn'] = true,
```
to the EXCLUDE table.

The second is the problem in `...xmf-dist/tex/luatex/luaotfload/fontloader-2016-06-16.lua:15598`.  I don't know why the spacing table is null when we are emitting characters from the fonts in `./../fonts` (via the `TTFONTS` environment variable).  It also seems to only happen when we *emit* `\{font}\char{cp}` from `GregorioRef.lua`.  Putting this same thing directly in the tex file does not cause this issue.

This problem can be "fixed" by editing fontloader-2016-06-16.lua and changing line 15598 to
```
    threshold=tfmdata.parameters.spacing and tfmdata.parameters.spacing.width or 65536*10
```

I'm not sure if this is a luaotfload bug or something we're doing wrong.

We can work around this by generating the tables beforehand, but this is a non-trivial undertaking.

I'm out of time for further investigation tonight.

**Comment by henryso** [comment]
I'm having trouble coming up with a minimum example that causes this error to happen.  I think this exercise is somewhat beyond me.

**Comment by eroux** [comment]
Well, if it's not possible, a non-minimal example wil do, I don't think it's a big problem...

**Comment by henryso** [comment]
Reported at lualatex/luaotfload#372.

**Comment by rpspringuel** [comment]
Thank you much for the work.  I was just expecting a few pointers on what to look at and how to logically break the code up, but you've gone much further than that.

**Comment by henryso** [comment]
I have not heard anything from the lualatex/luaotfload project for a few days, so I have a workaround in #1194 that unfortunately breaks the document build under 2015 (with a segfault).  See the pull request for more information.

**Comment by rpspringuel** [comment]
This was fixed in v2.8 of luaotfload so I'm closing this.
